cmd/util/ledger/migrations/core_contracts_migration.go

```diff
@@ -0,0 +1,325 @@
+	qc                string
+	epoch             string
+	dkg               string
+	stakingTable      string
+	lockedTokens      string
+	stakingProxy      string
+	stakingCollection string
+}
+
+func coreContractAddresses(chain flow.ChainID) networkAddresses {
+	switch chain {
+	case flow.Testnet:
+		return networkAddresses{
+			fungibleToken:     "9a0766d93b6608b7",
+			flowToken:         "7e60df042a9c0868",
+			contractAudits:    "8c5303eaa26202d6",
+			serviceAccount:    "8c5303eaa26202d6",
+			storageFees:       "8c5303eaa26202d6",
+			fees:              "912d5440f7e3769e",
+			qc:                "9eca2b38b18b5dfe",
+			epoch:             "9eca2b38b18b5dfe",
+			dkg:               "9eca2b38b18b5dfe",
+			stakingTable:      "9eca2b38b18b5dfe",
+			lockedTokens:      "95e019a17d0e23d7",
+			stakingCollection: "95e019a17d0e23d7",
+			stakingProxy:      "7aad92e5a0715d21",
+		}
+	case flow.Mainnet:
+		// TODO:
+		panic("TODO")
+	default:
+		panic(fmt.Errorf("unsupported chain: %s", chain))
+	}
+}
+
+func coreContractCodes(addresses networkAddresses) map[string]map[string][]byte {
+	codes := map[string]map[string][]byte{}
+
+	addCode := func(address, name string, code []byte) {
+		addressCodes, ok := codes[address]
+		if !ok {
+			addressCodes = map[string][]byte{}
+			codes[address] = addressCodes
+		}
+		addressCodes[name] = code
+	}
+
+	addCode(
+		addresses.contractAudits,
+		"FlowContractAudits",
+		coreContracts.FlowContractAudits(),
+	)
+	addCode(
+		addresses.serviceAccount,
+		"FlowServiceAccount",
+		coreContracts.FlowServiceAccount(
+			// fungibleTokenAddress:
+			addresses.fungibleToken,
+			// flowTokenAddress:
+			addresses.flowToken,
+			// flowFeesAddress:
+			addresses.fees,
+			// storageFeesAddress:
+			addresses.storageFees,
+		),
+	)
+	addCode(
+		addresses.storageFees,
+		"FlowStorageFees",
+		coreContracts.FlowStorageFees(
+			// fungibleTokenAddress:
+			addresses.fungibleToken,
+			// flowTokenAddress:
+			addresses.flowToken,
+		),
+	)
+	addCode(
+		addresses.fees,
+		"FlowFees",
+		coreContracts.FlowFees(
+			// fungibleTokenAddress:
+			addresses.fungibleToken,
+			// flowTokenAddress:
+			addresses.flowToken,
+		),
+	)
+	addCode(
+		addresses.qc,
+		"FlowClusterQC",
+		coreContracts.FlowQC(),
+	)
+	addCode(
+		addresses.dkg,
+		"FlowDKG",
+		coreContracts.FlowDKG(),
+	)
+	addCode(
+		addresses.epoch,
+		"FlowEpoch",
+		coreContracts.FlowEpoch(
+			// fungibleTokenAddress:
+			addresses.fungibleToken,
+			// flowTokenAddress:
+			addresses.flowToken,
+			// idTableAddress:
+			addresses.stakingTable,
+			// qcAddress:
+			addresses.qc,
+			// dkgAddress:
+			addresses.dkg,
+			// flowFeesAddress:
+			addresses.fees,
+		),
+	)
+	addCode(
+		addresses.stakingTable,
+		"FlowIDTableStaking",
+		coreContracts.FlowIDTableStaking(
+			// fungibleTokenAddress:
+			addresses.fungibleToken,
+			// flowTokenAddress:
+			addresses.flowToken,
+			// flowFeesAddress:
+			addresses.fees,
+			// latest:
+			true,
```

Comment:
I don't know what this boolean is

Reply:
https://github.com/onflow/flow-core-contracts/blob/842c892f75675133fa77720ac46ae0dfdd2ddceb/lib/go/contracts/contracts.go#L176-L177